### PR TITLE
Target validation during github build

### LINKED ISSUE #1902: Unified_ESP32_2400_TR target on rx_vtx
It looks like it should be Unified_ESP32_2400_RX?

Permalink: https://github.com/ExpressLRS/ExpressLRS/blob/bad7d1a6deaa1196311b04134c9878e82be6af43/src/hardware/targets.json#L536

### PR DESCRIPTION
**Pull request by pkendall64**
# Problem
Theres been multiple instances of wrong target definitions in the `targets.json` file, or in the `.ini` files, mostly by me!

# Solution
I got annoyed at myself with these silly little errors, so I wrote a validator and hooked it up to the build so hopefully we don't fall into this trap again!

And in the course of writing the validator I came across some other broken targets 😢 

Fixes #1902

### CODE REVIEW

**Review by JyeSmith — APPROVED** [review]

**Comment by JyeSmith** [comment]
Nice!  Looks to be doing its job and picking up errors... I might give it a sneaky test next PR 😄

File: src/python/targets_validator.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        if 'offset' not in stlink:
+            error(f'The "stlink" attribute for target "{vendor}.{type}.{devname}" must have a valid "offset"')
+        if 'bootloader' not in stlink:
+            error(f'The "stlink" attribute for target "{vendor}.{type}.{devname}" must have a valid "bootloader"')
+    # could check the existence of the bootloader file
+
+def validate_esp(vendor, type, devname, device):
+    if 'lua_name' not in device:
+        error(f'device "{vendor}.{type}.{devname}" must have a "lua_name" child element')
+    if len(device['lua_name']) > 16:
```
Comment (JyeSmith) [inline src/python/targets_validator.py:33]: I thought this was 15?

Comment (pkendall64) [reply src/python/targets_validator.py:33]: So did I, but theres a bunch at 16!

**Comment by pkendall64** [comment]
Fixes #1902

**Review by schugabe — APPROVED** [review]

**Pull request merged**